### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by mirovix**
1. Unit Test completed
2. Integration Test completed
3. Refactoring

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,14 +16,15 @@ find_package(catkin REQUIRED COMPONENTS
...
+find_package(GTest REQUIRED)
 find_package(PkgConfig)
 
 SET(CMAKE_BUILD_TYPE RelWithDebInfo)
 
 include_directories(${catkin_INCLUDE_DIRS} 
                     ${EIGEN3_INCLUDE_DIRS}
-					include)
+					include
+					include/${PROJECT_NAME}/)
```
Comment (PaoloFori) [inline CMakeLists.txt]: remove project name

**Review by PaoloFori — CHANGES_REQUESTED** [review]
this package with also the one of hann, blackman and flattop must depend on rosneuro_window and the package's name must be rosneuro_windows_hamming or hann or ...
Hence create new packages: rosneuro_windows, rosneuro_windows_hamming, rosneuro_windows_hann, rosneuro_windows_flattop, rosneuro_windows_blackman. For the code part refer to the repository on neurorobotics, but create new ones in rosneuro (set them private)

File: CMakeLists.txt (hunk 1 above)
Comment (mirovix) [reply CMakeLists.txt]: 1. Thanks for your detailed instructions. I was planning to create these new packages once your changes are approved. However, if there's a pressing need to do it sooner, please let me know, and I'll prioritize accordingly.

2. Project name removed from "include_directories".

Comment (PaoloFori) [reply CMakeLists.txt]: create the new packages in order to have a complete version of Rosneuro so that we can test it before making it public

**Review by PaoloFori — APPROVED** [review]

**Pull request merged**